actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -361,12 +362,20 @@ def render_partial(view, template)
             view._layout_for(*name, &block)
           end
 
+          partial_hints(template, content, locals) if render_hints
           content = layout.render(view, locals) { content } if layout
           payload[:cache_hit] = view.view_renderer.cache_hits[template.virtual_path]
           build_rendered_template(content, template)
         end
       end
 
+      def partial_hints(template, content, locals)
+        uuid = SecureRandom.uuid
+        content.prepend("\n<!-- start render: #{template.virtual_path}, uuid: #{uuid}, locals: #{locals.inspect} -->\n".html_safe)
```

Comment:
I'm not comfortable with inspecting the locals since there are a few locals that are internal and I don't think we should show them. Can we exclude those locals?

Reply:
Definitely. Updated to exclude the iterator and counter.